rbc/heavydb/column_list.py

```diff
@@ -18,20 +18,62 @@ class HeavyDBColumnListType(Type):
 
     """
 
+    @property
+    def element_type(self):
+        return self[0][0]
+
+    @property
+    def buffer_extra_members(self):
+        if self.element_type.tostring() == 'TextEncodingDict':
+            return ('i8** string_dict_proxy_',)
+        return ()
+
     @property
     def __typesystem_type__(self):
-        element_type = self[0][0]
-        ptrs_t = element_type.pointer().pointer().params(name='ptrs')
+        ptrs_t = self.element_type.pointer().pointer().params(name='ptrs')
         length_t = Type.fromstring('int64 length')
         size_t = Type.fromstring('int64 size')
-        return Type(ptrs_t, length_t, size_t).params(
+        extra_members = tuple(map(Type.fromobject, self.buffer_extra_members))
+        return Type(ptrs_t, length_t, size_t, *extra_members).params(
             NumbaPointerType=HeavyDBColumnListNumbaType).pointer()
 
 
+@extending.intrinsic
+def heavydb_columnlist_getitem(typingctx, lst, idx):
+    from .column import HeavyDBColumnType
+
+    members = lst.dtype.members
+    is_text_encoding_dict = False
+    if len(members) == 4:
+        is_text_encoding_dict = True
+        T = Type.fromstring('TextEncodingDict')
+    else:
+        T = Type.fromnumba(lst.dtype.members[0].dtype.dtype)
```

Comment:
Perhaps there's a better way to detect if the columnlist is of type `TextEncodingDict`.